drivers/can/can_mcan.c

```diff
@@ -828,6 +828,17 @@ static int can_mcan_get_free_std(volatile struct can_mcan_std_filter *filters)
 	return -ENOSPC;
 }
 
+int can_mcan_get_max_filters(const struct device *dev, enum can_ide id_type)
+{
+	ARG_UNUSED(dev);
+
+	if (id_type == CAN_STANDARD_IDENTIFIER) {
+		return NUM_STD_FILTER_ELEMENTS;
+	} else {
+		return NUM_EXT_FILTER_ELEMENTS;
+	}
```

Comment:
In the STM32 driver we can limit the amount of actually used filters via Kconfig to save some RAM. We have to use `_DATA` instead of `_ELEMENTS` macros here:

```suggestion
	if (id_type == CAN_STANDARD_IDENTIFIER) {
		return NUM_STD_FILTER_DATA;
	} else {
		return NUM_EXT_FILTER_DATA;
	}
```

Otherwise we will may get an overflow here:
https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/can/can_mcan.h#L175-L178

Because of:
https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/can/can_mcan.h#L31-L37

If the number of buffers was limited.

Reply:
Good catch! Thank you.